ros/src/tl_detector/tl_detector.py

```diff
@@ -173,28 +175,16 @@ def get_light_state(self, light):
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
 
         """
-        if(not self.has_image):
-            self.prev_light_loc = None
-            return False
-
         if self.get_light == 'oracle':
-            if self.record_path != '' and time.time() - self.record_time > RECORD_SLEEP:
-                cv_image = self.bridge.imgmsg_to_cv2(self.camera_image, "bgr8")
-                # Make a unique identifier from a time stamp (good enough for a unique identifier and more readable than
-                # uuid).
-                now = datetime.datetime.utcnow()
-                format_arg = now.year, now.month, now.day, now.hour, now.minute, now.second, now.microsecond
-                fname = 'TL_{}_{:02}_{:02}_{:02}_{:02}_{:02}_{:06}.jpg'.format(*format_arg)
-                subfolder = str(light.state) + '_' + LIGHT_STATES[light.state].lower()
-                path = os.path.join(self.record_path, subfolder, fname)
-                cv2.imwrite(path, cv_image)
-                self.record_time = time.time()
             return light.state
-
-        cv_image = self.bridge.imgmsg_to_cv2(self.camera_image, "rgb8")
-
-        #Get classification
-        return self.light_classifier.get_classification(cv_image)
+        elif self.get_light == 'detect':
+            if self.has_image:
+                self.prev_light_loc = None
+                cv_image = self.bridge.imgmsg_to_cv2(self.camera_image, "bgr8")
```

Comment:
As for the previous pull request, here we need `rgb8` instead of `bgr8` otherwise red and yellow light can't be detected.